### PR TITLE
Trying to fix the .m3u decoding problem

### PR DESCRIPTION
**Pull request by gjelsas**
This is a easy fix for this [Issue](https://github.com/beedaddy/radiorec/issues/15#issue-737895769). The current Code is not storing anything in `streamurl`

### CODE REVIEW

**Comment by monofox** [comment]
Are you sure it is working for all kind of variance of m3u playlists? The for loop was there for a reason. I mean if you have an advanced formatted m3u playlist, multiple first lines will have a `#`.
E.g.
```
#EXTM3U
#EXTINF:-1,SWR2
http://swr-swr2-live.cast.addradio.de/swr/swr2/live/mp3/256/stream.mp3

```
Tested with your changes:
```
[I]  22:18:32  ✔  ~/.p/gjelsas-radiorec   master  $  python3 radiorec.py record swr2 12
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): mp3-live.swr.de:80
DEBUG:urllib3.connectionpool:http://mp3-live.swr.de:80 "GET /swr2_m.m3u HTTP/1.1" 200 96
Traceback (most recent call last):
  File "radiorec.py", line 195, in <module>
    main()
  File "radiorec.py", line 192, in main
    args.func(args)
  File "radiorec.py", line 129, in record
    streamurl = tmpstr
UnboundLocalError: local variable 'tmpstr' referenced before assignment
```

**Comment by gjelsas** [comment]
Hmm, strange, maybe mine where resolved differently? I don't see a reason why that would be the case... I'm currently using radiorec on Ubuntu 20.04... 
When debugging radiorec, the streamurl variable was empty and therefore throwing the exception described in the title if the corresponding issue... 
Could this be a dependency issue for me?

**Comment by gjelsas** [comment]
I changed the code to store the HTTPResponse object in remotefile and iterate over the lines
And I added my favorite radio station...

**Comment by gjelsas** [comment]
Added the mentioned checks. Maybe writing some tests would be good in the long run...

File: radiorec.py [hunk 1]
```diff
@@ -123,15 +125,30 @@ def record(args):
...
+        if remotefile.status != 200:
+            print('The URL of the station is somehow faulty! Check' + args.station + ' in the Settings!')
+            sys.exit()
+        else:
+            for line in remotefile.data.decode().split():
+                if not line.startswith('#') and len(line) > 1 and line.endswith('mp3'):
+                    tmpstr = line
                     break
-        streamurl = tmpstr
+        if not len(tmpstr) > 1:
```
Comment (monofox) [inline radiorec.py:141]: Python allows to just check on `if tmpstr` (see https://onlinegdb.com/3tujtNGvK)

File: radiorec.py [hunk 2]
```diff
@@ -123,15 +125,30 @@ def record(args):
...
         pool = urllib3.PoolManager()
-        with pool.request('GET',streamurl) as remotefile:
-            for line in remotefile:
-                if not line.decode('utf-8').startswith('#') and len(line) > 1:
-                    tmpstr = line.decode('utf-8')
+        try:
+            remotefile = pool.request('GET', streamurl)
+        except MaxRetryError:
+            print('The URL of the station is not found! Check' + args.station + ' in the Settings!')
+            sys.exit()
```
Comment (monofox) [inline radiorec.py:132]: Errors should be indicated with a positive number (e.g. `sys.exit(1)`) as this status can be checked by other scripts.

File: radiorec.py [hunk 3]
```diff
@@ -123,15 +125,30 @@ def record(args):
...
         verboseprint('Seems to be an M3U playlist. Trying to parse...')
         pool = urllib3.PoolManager()
-        with pool.request('GET',streamurl) as remotefile:
-            for line in remotefile:
-                if not line.decode('utf-8').startswith('#') and len(line) > 1:
-                    tmpstr = line.decode('utf-8')
+        try:
+            remotefile = pool.request('GET', streamurl)
+        except MaxRetryError:
+            print('The URL of the station is not found! Check' + args.station + ' in the Settings!')
```
Comment (monofox) [inline radiorec.py:131]: Errors should be indicated to stderr, not stdout. With this you're able to filter out the debug messages or info messages if you just would like to extract the errors (e.g. for sending automatically reports via scripts).
So i recommend to get a log object (via `logging.getLogger(__name__)`) and log it with `error` function.
E.g.:
`logging.getLogger(__name__).error('The URL of the station is somehow faulty! Check' + args.station + ' in the Settings!')`

Would do this for this line as well as line 134.

Comment (monofox) [reply radiorec.py:131]: +line 142 + line 150

File: radiorec.py [hunk 4]
```diff
@@ -123,15 +125,30 @@ def record(args):
...
-        streamurl = tmpstr
+        if not len(tmpstr) > 1:
+            print('Could not find a mp3 stream')
+            sys.exit()
+        else:
+            streamurl = tmpstr
 
     verboseprint(print_time() + " ... Stream URL: " + streamurl)
     target_dir = os.path.expandvars(settings['GLOBAL']['target_dir'])
+    if not os.path.isdir(target_dir + os.sep):
```
Comment (monofox) [inline radiorec.py:149]: why `+ os.sep`? 
Furthermore we could (at least) try to create the directory with `os.makedirs` prior this check with `exist_ok=True`

**Review by monofox — COMMENTED** [review]
Thank you very much for your contribution. I'd a look on it and proposed some improvements to have a better control if the application is used by an automatism / script.

**Comment by gjelsas** [comment]
I included your mentions! Hope its okay now! Frohe Weihnachten!

**Comment by beedaddy** [comment]
@gjelsas Thanks for your contribution. For me it looks good (although I couldn't test it thoroughly).

**Pull request merged**